src/main/kotlin/it/pagopa/wallet/domain/details/WalletDetails.kt

```diff
@@ -3,4 +3,6 @@ package it.pagopa.wallet.domain.details
 /** Extensible interface to handle multiple wallet details typologies, such as CARDS */
 sealed interface WalletDetails {
     val type: WalletDetailsType
+
+    fun toDocument(): it.pagopa.wallet.documents.wallets.details.WalletDetails
```

Comment:
Change interface declaration using generic as follow:
`sealed interface WalletDetails<T> where T : it.pagopa.wallet.documents.wallets.details.WalletDetails`
and then, this method as:
```suggestion
    fun toDocument(): T
```

Reply:
Fixed in cdb1ad8